merkletree/proof.go

```diff
@@ -16,6 +23,75 @@ type ProofNode interface {
 	Commitment() []byte
 }
 
+func computeLeafHash(ap *AuthenticationPath,
+	leafIndex, leafCommitment []byte, leafLevel int, isLeafEmpty bool) (leafHash []byte) {
+	if isLeafEmpty {
```

Comment:
If we're passing in ``ap`` to ``computeLeafHash()`` couldn't we use the index, commitment, level etc from the ``ap.Leaf`` directly? On the other hand, I think this would be a similar issue to what we discussed in https://github.com/coniks-sys/coniks-go/commit/d46c806796a4f58b038fc1ee8794b2bccfd8e8aa#commitcomment-18437585, so I don't know if we should be passing in any auth path pointer at all.